src/gryphlifehandler.js

```diff
@@ -0,0 +1,117 @@
+import * as http from "http";
+import * as ExcelJS from "exceljs";
+import * as url from "url";
+import cors from "cors";
+import dotenv from "dotenv";
+// import { Client, TextChannel } from 'discord.js';
+
+dotenv.config();
+
+const hostname = "127.0.0.1";
+const port = 5000;
+//Excel File Path
+const filePath = "";
+// const channelId = '';
+
+async function processExcelData(client, data) {
+    console.log("Data received:", data);
+
+    // Your Excel file path
+
+    // Load the workbook and select the first worksheet
+    const workbook = new ExcelJS.default.Workbook();
+    await workbook.xlsx.readFile(filePath);
+    const sheet = workbook.getWorksheet(1);
+
+    // Find the column with the header "In GryphLife"
+    let gryphLifeColNum = null;
+    sheet.eachRow((row, rowNumber) => {
+        if (rowNumber === 1) {
+            // Assuming the headers are in the first row
+            row.eachCell((cell, colNumber) => {
+                if (cell.value === "In GryphLife") {
+                    gryphLifeColNum = colNumber;
+                }
+            });
+        }
+    });
+
+    // If the "In GryphLife" column is found
+    if (gryphLifeColNum !== null) {
+        // Iterate through the received data
+        for (let member of data) {
+            // Assume names are in column A, find the first empty row in that column
+            let row = 1;
+            while (sheet.getCell(`A${row}`).value !== null) {
+                row += 1;
+            }
+
+            // Write the name and email to the first empty cell in column A and B respectively
+            const name = member.Name;
+            const email = member.email;
+            if (name) {
+                sheet.getCell(`A${row}`).value = name;
+            }
+            if (email) {
+                sheet.getCell(`B${row}`).value = email;
+            }
+
+            // Set the corresponding "In GryphLife" column cell to "yes"
+            sheet.getCell(`E${row}`).value = "yes";
+        }
+    } else {
+        console.error("Column 'In GryphLife' not found");
+    }
+
+    // Save the workbook
+    await workbook.xlsx.writeFile(filePath);
```

Comment:
This entire section is redundant. We use a singleton class now to manage the entire spreadsheet. Your changes can be made by pushing your changes into a data_queue stack. See [here](https://github.com/GryphonRacingFSAE/Discord-Bot/blob/6c5cb59c45417faa52f9147ae5a39b5ab01c0cf2/src/vertification.ts#L57-L74) for more.